controller/topology_builder.go

```diff
@@ -24,31 +24,31 @@ type gatewayAPITopologyBuilder struct {
 }
 
 func (t *gatewayAPITopologyBuilder) Build(objs Store) *machinery.Topology {
-	gatewayClasses := lo.FilterMap(lo.Values(objs[schema.GroupKind{Group: gwapiv1.GroupVersion.Group, Kind: "GatewayClass"}]), func(obj RuntimeObject, _ int) (*gwapiv1.GatewayClass, bool) {
+	gatewayClasses := lo.FilterMap(lo.Values(objs[machinery.GatewayClassGroupKind]), func(obj RuntimeObject, _ int) (*gwapiv1.GatewayClass, bool) {
 		gc, ok := obj.(*gwapiv1.GatewayClass)
 		if !ok {
 			return nil, false
 		}
 		return gc, true
```

Comment:
It doesn't belong to this PR exactly, but I guess these filter maps could be simplified.

```suggestion
	gatewayClasses := lo.FilterMap(lo.Values(objs[machinery.GatewayClassGroupKind]), func(obj RuntimeObject, _ int) (*gwapiv1.GatewayClass, bool) {
		return obj.(*gwapiv1.GatewayClass)
```

Reply:
Done for the existing wrappers in https://github.com/Kuadrant/policy-machinery/pull/17/commits/112499121c28f3700e4234e1f0413bd76473043d#diff-cf66b18260b0367d1ba0a51dd11004e8d9a5ec5c37c31c92be70630305bc7737

Reply:
The above gives an error `not enough return values` on the IDE, so the above won't directly unless I'm missing something 🤔 

The below would work I think without hardcoding the boolean returns like `return obj.(*gwapiv1.GatewayClass) false|true`.  
```suggestion
	gatewayClasses := lo.FilterMap(lo.Values(objs[machinery.GatewayClassGroupKind]), func(obj RuntimeObject, _ int) (*gwapiv1.GatewayClass, bool) {
		gc, ok := obj.(*gwapiv1.GatewayClass)
		return gc, ok
```
What do you think?

Reply:
You're right. We need to lines at least.